### PR TITLE
Handle Register Sessions

### PR DESCRIPTION
**Pull request by Drish-xD**
- Added a platform - `no-platform`
- Allow edit to register sessions by removing the required property of auth field if `isRedirection` is false.
- Remove `ID,PH`, `ID,PH,DOB` auth options
- Remove Signup `sign-in with forgot id`

Preview: https://handle-registration.dauyvuo7dvzau.amplifyapp.com/

Tested: Create & edit of registration sessions.

### CODE REVIEW

File: src/types/api.types.ts [hunk 1]
```diff
@@ -48,7 +48,7 @@ export const sessionSchema = z.object({
   meta_data: metaDataSchema,
   name: z.string(),
   owner_id: z.string(),
-  platform: z.string(),
+  platform: z.string().nullable(),
```
Comment (suryabulusu) [inline src/types/api.types.ts:51]: i see the problem with having platform as null now... maybe we should just give an option "No Platform" ? Not sure

File: src/app/session/[type]/Steps/helper.ts [hunk 2]
```diff
@@ -283,6 +278,9 @@ export const handleBatchFields = (
     fieldsSchema.parentBatch.hide = false;
   }
 
+  if (value === Platform.NoPlatform) {
+    form.setValue('isRedirection', false, { shouldDirty: true });
```
Comment (suryabulusu) [inline src/app/session/[type]/Steps/helper.ts:282]: if value changes, set redirection to true

**Review by suryabulusu — APPROVED** [review]
looks good. few comments

File: src/types/enums.ts [hunk 3]
```diff
@@ -66,10 +67,11 @@ export const GroupShortName: Record<Group, string> = {
 
 export enum AuthType {
```
[the remark this answers is not in the document]
Comment (Drish-xD) [reply src/types/enums.ts:68]: It's fetched from authgroup table itself.

File: src/app/session/[type]/Steps/helper.ts (hunk 2 above)
Comment (Drish-xD) [reply src/app/session/[type]/Steps/helper.ts:282]: Added

**Pull request merged**